src/plugins/mission_raw/mission_raw_impl.cpp

```diff
@@ -22,16 +22,36 @@ void MissionRawImpl::init()
         MAVLINK_MSG_ID_MISSION_ACK,
         std::bind(&MissionRawImpl::process_mission_ack, this, _1),
         this);
+
+    _parent->register_mavlink_message_handler(
+        MAVLINK_MSG_ID_MISSION_CURRENT,
+        std::bind(&MissionRawImpl::process_mission_current, this, _1),
+        this);
+
+    _parent->register_mavlink_message_handler(
+        MAVLINK_MSG_ID_MISSION_ITEM_REACHED,
+        std::bind(&MissionRawImpl::process_mission_item_reached, this, _1),
+        this);
 }
 
+void MissionRawImpl::enable() {}
+
+void MissionRawImpl::disable() {}
+
 void MissionRawImpl::deinit()
 {
+    _parent->unregister_timeout_handler(_timeout_cookie);
     _parent->unregister_all_mavlink_message_handlers(this);
 }
 
-void MissionRawImpl::enable() {}
-
-void MissionRawImpl::disable() {}
+void MissionRawImpl::reset_mission_progress()
+{
+    std::lock_guard<std::recursive_mutex> lock(_mission_data.mutex);
+    _mission_data.last_current_mavlink_mission_item = -1;
+    _mission_data.last_reached_mavlink_mission_item = -1;
+    _mission_data.last_current_reported_mission_item = -1;
+    _mission_data.last_total_reported_mission_item = -1;
```

Comment:
Isn't `current_mavlink` and `current_reported` the same for `mission_raw`?

Reply:
`current_mavlink` gives the number of MAV_CMD we sent and `current_reported` is mapping every command for a given waypoint right ?

IMO this is interesting to keep this mapping. My use case is that I want to generate a Mission but to customize like I want so I send `MissionRaw` rather than `Mission` but without to change the logic of following progress. To be correct we may add a `subscribe_malvink_progress` that returns the progress of the mavlink command (if my question above is correct)

Reply:
Aha. I don't think the mapping makes sense because there might be more mission item types that we should count that we currently are not counting.

Since this is raw I would strongly suggest that also the progress is raw. Otherwise it gets confusing.

Reply:
ok yes I understand that it could not match all use cases. I'll do this modification and manage this on my side. So do we agree I have to remove all the `*_mission_item_*` related  and that progress/upload/download/indice/etc are defined by a `*_mavlink_message_item_*` or another name like this. Otherwise it would be a bit confusing to have "mission_item" name

Reply:
ok sounds good.